compiler/rustc_typeck/src/check/writeback.rs

```diff
@@ -195,6 +195,48 @@ impl<'cx, 'tcx> WritebackCx<'cx, 'tcx> {
         }
     }
 
+    // (ouz-a #100498): Normally `[T] : std::ops::Index<usize>` should be normalized
+    // into [T] but currently `Where` clause stops the normalization process for it,
+    // here we check if the expr is `LangItem::Index` and index_ty is `usize` if so we don't
+    // `fix_index_builtin_expr`.
+    fn is_builtin_index(
+        &mut self,
+        e: &hir::Expr<'_>,
+        base: Ty<'tcx>,
+        index: Ty<'tcx>,
+        base_span: Span,
+        index_span: Span,
+    ) -> bool {
+        let mut ret = false;
+        if (&base).builtin_index().is_some() {
+            let resolved_base_ty = self.resolve(base, &base_span);
+            let resolved_index_ty = self.resolve(index, &index_span);
+            self.tcx().infer_ctxt().enter(|infcx| {
+                let mut selection_context = SelectionContext::new(&infcx);
+
+                let def_id = self.tcx().require_lang_item(LangItem::Index, Some(e.span));
```

Comment:
nit: We could probably use a `let else` here, and just `return false;` if this lang item is missing rather than erroring.